### PR TITLE
Fix strict branch protection

### PR DESCRIPTION
**Pull request by kfcampbell**
Strict protection on v4 branch protection doesn't work at the moment. This PR makes minor code changes and removes the diff suppression func in order to correct strict branch protection. 

The research to get to this point is chronicled in #880. 

The TL;DR is that by removing the diff suppression and making a minor code change to the branch protection processing, we can restore strict branch protection with the graphQL client.

### CODE REVIEW

**Comment by kfcampbell** [comment]
Alright I've spent more time here and done an expanded testing matrix on both the current release and this feature branch. The results are as follows:

### main branch

- branch protection, without required status checks
	- same weird state with the hidden checked box
	- churn: initial push_restrictions = [] then nothing
- branch protection, with required status checks, strict = false
	- initial apply is broken, no required status checks, same weird state with the hidden checked box
	- churn: initial push_restrictions = [] then nothing, still broken
- branch protection, with required status checks, strict = false, with contexts
	- initial apply works
	- churn: initial push_restrictions = [] then nothing
	- if contexts are present but empty, same broken behavior as above is exhibited
- branch protection, with required status checks, strict = true
	- initial apply is broken, no required status checks, same weird state with the hidden checked box
	- churn: initial push_restrictions = [] then nothing, still broken
- branch protection, with required status checks, strict = true, with contexts
	- initial apply works
	- churn: initial push_restrictions = [] then nothing
	- if contexts are present but empty, same broken behavior as above is exhibited
	
### bugfix branch

- branch protection, without required status checks
	- same weird state with the hidden checked box
	- churn: initial push_restrictions = [] then nothing
- branch protection, with required status checks, strict = false
	- initial apply works
	- churn: initial push_restrictions = [] and contexts = [] then nothing
- branch protection, with required status checks, strict = false, with contexts
	- initial apply works
	- churn: initial push_restrictions = [] then nothing
	- if contexts are present but empty, there's churn: initial push_restrictions = [] and contexts = [] then nothing
- branch protection, with required status checks, strict = true
	- initial apply works
	- churn: initial push_restrictions = [] and contexts = [] then nothing
- branch protection, with required status checks, strict = true, with contexts
	- initial apply works
	- churn: initial push_restrictions = [] then nothing

In addition, @jcudit and I have done some internal investigation and it looks like the weird behavior with the hidden checkbox already checked is by design, or at least it's been that way from the beginning. 

TL;DR this PR is ready for review! The bug is fixed on this branch and there's no regression on #572.

**Review by jcudit — APPROVED** [review]
